### PR TITLE
[Imagecopy] Validation on resource group; Improve logging

### PR DESCRIPTION
**Pull request by qwordy**
1.
Resolve https://github.com/Azure/azure-cli-extensions/issues/1584
`--cleanup` : Include this switch to delete temporary resources upon completion.
`--temporary-resource-group-name`    : Resource Group name where temporary storage account will be created.  Default: image-copy-rg.
If `--cleanup` is set and `--temporary-resource-group-name` is set an existing resource group, this resource group will be deleted, even the resources created by the user before.

Now it is forbidden to use existing resource group name is `--cleanup` is set.

2.
Add more logs. Print running commands.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Image copy

File: src/image-copy/azext_imagecopy/__init__.py [hunk 1]
```diff
@@ -47,7 +47,7 @@ def load_arguments(self, _):
             c.argument('timeout', options_list=['--timeout'], type=int, default=3600,
                        help='Time in seconds for the copy operation to finish. Increase this time if '
                        'you are going to copy large images (disks) like 512GB or more.')
-            c.argument('temporary_resource_group_name', options_list=['--temporary_resource_group_name'],
+            c.argument('temporary_resource_group_name', options_list=['--temporary-resource-group-name'],
```
Comment (arrownj) [inline src/image-copy/azext_imagecopy/__init__.py:50]: break change ?

Comment (qwordy) [reply src/image-copy/azext_imagecopy/__init__.py:50]: The old name is wrong. It is a breaking change, but it should be updated.

File: src/image-copy/azext_imagecopy/cli_utils.py [hunk 2]
```diff
@@ -86,3 +88,9 @@ def get_storage_account_id_from_blob_path(cmd, blob_path, resource_group, subscr
         namespace='Microsoft.Storage', type='storageAccounts', name=storage_account_name)
 
     return storage_account_id
+
+
+def format_cmd(cmd: List[str]):
+    if len(cmd) > 3:
+        return 'Running: az ' + ' '.join(cmd[3:])
```
Comment (jiasli) [inline src/image-copy/azext_imagecopy/cli_utils.py:95]: Is this safe on all platforms with different entry scripts? If so, add some comments, otherwise using the magic number `3` in `cmd[3:]` may cause trouble.

File: src/image-copy/azext_imagecopy/cli_utils.py [hunk 3]
```diff
@@ -86,3 +88,9 @@ def get_storage_account_id_from_blob_path(cmd, blob_path, resource_group, subscr
...
+    return 'Running: ' + ' '.join(cmd)
```
Comment (jiasli) [inline src/image-copy/azext_imagecopy/cli_utils.py:96]: This may be confusing when `cmd`'s items contain spaces or quotes. Say `hello there` will become `az hello there`, instead of `az "hello there"`. `"hello there"` will become `az "hello there"`, instead of `az "\"hello there\""`.

File: src/image-copy/azext_imagecopy/create_target.py [hunk 4]
```diff
@@ -37,6 +39,7 @@ def create_target_image(cmd, location, transient_resource_group_name, source_typ
                                    '--sku', 'Standard_LRS'],
                                   subscription=target_subscription)
 
+    logger.warning(format_cmd(cli_cmd))
```
Comment (tamirkamara) [inline src/image-copy/azext_imagecopy/create_target.py:42]: Looks like you're printing every command. What about doing it just 1 time inside of the run_cli_command function?
BTW, that function is already printing it out, but as a "debug" message... why do we need it twice?

File: src/image-copy/azext_imagecopy/custom.py [hunk 5]
```diff
@@ -19,14 +21,24 @@
 # pylint: disable=too-many-branches
 def imagecopy(cmd, source_resource_group_name, source_object_name, target_location,
               target_resource_group_name, temporary_resource_group_name='image-copy-rg',
-              source_type='image', cleanup='false', parallel_degree=-1, tags=None, target_name=None,
+              source_type='image', cleanup=None, parallel_degree=-1, tags=None, target_name=None,
               target_subscription=None, export_as_snapshot='false', timeout=3600):
+    # If --cleanup is set, forbid using an existing temporary resource group name.
+    # It is dangerous to clean up an existing resource group.
+    if cleanup:
```
Comment (tamirkamara) [inline src/image-copy/azext_imagecopy/custom.py:28]: There's a person that also had an issue when using the same source & target resource group name with cleanup=true. That essentially deleted the source group which included a bunch of other VMs.
Another check I had in mind is not allowing same source/target (on the same subscription) when cleaning up. But that might be for another PR...

File: src/image-copy/azext_imagecopy/cli_utils.py (hunk 2 above)
Comment (qwordy) [reply src/image-copy/azext_imagecopy/cli_utils.py:95]: You are right. I haven't tested in other platforms except Powershell.

File: src/image-copy/azext_imagecopy/cli_utils.py (hunk 3 above)
Comment (qwordy) [reply src/image-copy/azext_imagecopy/cli_utils.py:96]: Yes. The returned string may not recover the intent of cmd variable.

**Comment by qwordy** [comment]
I will split this PR into two PRs.

File: src/image-copy/azext_imagecopy/create_target.py (hunk 4 above)
Comment (qwordy) [reply src/image-copy/azext_imagecopy/create_target.py:42]: Almost. But I skip `wait_for_blob_copy_operation` because `storage blob show` is called repeatedly.
The reason I make the change is that I want to print CLI command that can be executed directly.

File: src/image-copy/azext_imagecopy/custom.py (hunk 5 above)
Comment (qwordy) [reply src/image-copy/azext_imagecopy/custom.py:28]: Do you mean temporary resource group and target resource group?

Comment (tamirkamara) [reply src/image-copy/azext_imagecopy/custom.py:28]: sorry, I confused it a bit. 
I meant the source and the temporary one. There was an incident where a user gave the same same name for the temp one as the source and the process deleted many production resources.

Comment (qwordy) [reply src/image-copy/azext_imagecopy/custom.py:28]: Source resource group is an existing resource group, so this problem can be covered by this piece of code.
